include/crm/common/xml.h

```diff
@@ -395,4 +395,90 @@ void crm_xml_sanitize_id(char *id);
 void crm_xml_set_id(xmlNode *xml, const char *format, ...)
     __attribute__ ((__format__ (__printf__, 2, 3)));
 
+
+/*!
+ * \brief Base for directing lib{xml2,xslt} log into standard libqb backend
+ *
+ * This macro implements the core of what can be needed for directing
+ * libxml2 or libxslt error messaging into standard, preconfigured
+ * libqb-backed log stream.
+ *
+ * It's a bit unfortunate that libxml2 (and more sparsely, also libxslt)
+ * emits a single message by chunks (location is emitted separatedly from
+ * the * message itself), so we have to take the effort to combine these
+ * chunks back to single message.  Whether to do this or not is driven
+ * with \p dechunk toggle.
+ *
+ * The form of a macro was chosen for implicit deriving of __FILE__, etc.
+ * and also because static dechunking buffer should be differentiated per
+ * library (here we assume different functions referring to this macro
+ * will not ever be using both at once), preferably also per-library
+ * context of use to avoid clashes altogether.
```

Comment:
I'd rather have this as a function -- the main reason is that macros/inlines may require a bump in the library's shared object version if their implementation changes significantly, while only the signature matters for functions. And of course complex macros can be more susceptible to multiple evaluation of their arguments.

To handle function/file/line, we generally add them to the function arguments (often with a "_full" or "_alias" function name), then define a macro to call that function.

To handle the separate buffers, just pass it in.